src/index.tsx

```diff
@@ -105,14 +105,16 @@ startMathJax();
 serviceWorker.register()
   .then((registration) => {
     app.services.serviceWorker = registration;
-
-    if (registration && (registration.waiting || registration.installing)) {
+    const serviceWorkerController = window.navigator.serviceWorker.controller;
+    if (registration && (registration.waiting || registration.installing) && serviceWorkerController) {
```

Comment:
`&& serviceWorkerController` could be before `(registration.waiting || registration.installing)`